### ISSUE TITLE
Write example for taproot psbt APIs

### ISSUE THREAD

**Issue by sanket1729** [rust-bitcoin/rust-bitcoin#893]
Similar to #892, but that spends a taproot spend. This is more challenging because 
- Taproot psbt involves encoding a TapTree that is non-trivial to understand/use. 
- There is no spec to compare against. This would have to be checked against a bitcoin-core node to see if the spends and intermediate spends are valid.

**Comment by dunxen** [comment]
I can take a look at this one to get more familiar with TapTrees. Doesn't necessarily depend on a PR for #892, I understand.

**Comment by tcharding** [comment]
Correct, there is no dependency on #892, you can take a look at #940 for inspiration if you like.

**Comment by dunxen** [comment]
I've decided to write three different spending scenarios for this workflow:

1. Via key path with unspendable script path (à la BIP86) - Just a single-sig wallet
1. Via key path with spendable script path - Inheritance scheme where the benefactor can spend with just their key (key path)
1. Via script path with spendable key path - Inheritance scheme where the beneficiary can spend after an absolute timelock with their key (script path)

I've completed the first and will work on the remaining two and split them up into separate commits in the PR.
